lib/mix_unused/analyzers/unreachable/usages/amqpx_consumers_discovery.ex

```diff
@@ -0,0 +1,20 @@
+defmodule MixUnused.Analyzers.Unreachable.Usages.AmqpxConsumersDiscovery do
+  @moduledoc """
+  Discovers the consumers configured for the [amqpx library](https://hex.pm/packages/amqpx).
+  """
+
+  @behaviour MixUnused.Analyzers.Unreachable.Usages
+
+  @impl true
+  def discover_usages(_context) do
+    app = Mix.Project.config()[:app]
+
+    for %{handler_module: module} <- Application.get_env(app, :consumers, []) do
+      [
+        {module, :setup, 1},
+        {module, :handle_message, 3}
+      ]
+    end
+    |> List.flatten()
```

Comment:
```suggestion
    Enum.flat_map(Application.get_env(app, :consumers, []), fn %{handler_module: module} ->
      [
        {module, :setup, 1},
        {module, :handle_message, 3}
      ]
    end)
```

Reply:
In general I like to take advantage of the filtering feature of for-comprehension.

In your change for example there is a MatchError if the config has not the `handler_module`.
But I admit that `List.flatten()` is not so smart, so it could also be rewritten in this way:

```
for %{handler_module: module} <- Application.get_env(app, :consumers, []),
    mfa <- [{module, :setup, 1}, {module, :handle_message, 3}],
    do: mfa
```